Kernel/Arch/riscv64/SmapDisabler.cpp

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2023, Sönke Holz <sholz8530@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/Arch/SmapDisabler.h>
+#include <Kernel/Arch/riscv64/CSR.h>
+
+namespace Kernel {
+
+SmapDisabler::SmapDisabler()
```

Comment:
Future PR should probably rename this to something more generic than the x86 feature